pkg/detectors/rabbitmq/rabbitmq.go

```diff
@@ -0,0 +1,83 @@
+package rabbitmq
+
+import (
+	"context"
+	"net/url"
+	"regexp"
+	"strings"
+
+	amqp "github.com/rabbitmq/amqp091-go"
+
+	"github.com/trufflesecurity/trufflehog/v3/pkg/detectors"
+	"github.com/trufflesecurity/trufflehog/v3/pkg/pb/detectorspb"
+)
+
+type Scanner struct{}
+
+// Ensure the Scanner satisfies the interface at compile time.
+var _ detectors.Detector = (*Scanner)(nil)
+
+var (
+	keyPat = regexp.MustCompile(`\b(?:amqp:)?\/\/[\S]{3,50}:([\S]{3,50})@[-.%\w\/:]+\b`)
+)
+
+// Keywords are used for efficiently pre-filtering chunks.
+// Use identifiers in the secret preferably, or the provider name.
+func (s Scanner) Keywords() []string {
+	return []string{"amqp"}
+}
+
+// FromData will find and optionally verify RabbitMQ secrets in a given set of bytes.
+func (s Scanner) FromData(ctx context.Context, verify bool, data []byte) (results []detectors.Result, err error) {
+	dataStr := string(data)
+
+	matches := keyPat.FindAllStringSubmatch(dataStr, -1)
+
+	for _, match := range matches {
+		urlMatch := match[0]
```

Comment:
suggestion: We should add a len check here for match to prevent any IOOR panics.

if len(match) < 3 {
    continue
}